Add get assistance page

puneetlath opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
This PR adds a new `Get Assistance` page (shown below) which can be accessed by clicking the new question mark icon in the header of any page in the workspace settings. This page gives the option to chat with concierge or request a call with a guide. A summary of what was added/changed: 
1. Updated the header component to have the option to show the get assistance button (question mark) 
2. Updated every page of the workspace to show that button in its header
3. Added a new get assistance page, which is accessed by clicking that question mark button
4. Added constants for the various guides call Task IDs we'll associate with the guides call, depending on which page it is triggered from 
5. Added the functionality of storing a new key `lastAccessedWorkspace` in Onyx that is updated whenever a workspace is loaded (this is because there are scenarios where we can't get the workspace directly, context [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1640901164291600))
6. Removed the `InboxCallButton` component, which is no longer used anywhere in the workspace
7. Updated the request a call page to navigate back to the previous page after a call is requested, instead of always navigating to the Concierge DM
8. Updated the request a call page to use the `lastAccessedWorkspace` for the policy ID to associate with the guides job if it is available, otherwise fall back on the personal policy ID
9. Refactored the `WorkspaceSection` component to be a more generic `Section` component that can be used outside the workspace
10. Used that `Section` component on the new `GetAssistance` page and updated the `RequestCall` page to use it as well
 
### Fixed Issues
<!---
Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing.
Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the issue; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/6719
$ https://github.com/Expensify/Expensify/issues/184190

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/6719
$ https://github.com/Expensify/Expensify/issues/184190

### Tests
<!--- 
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image 
--->
1. Click through every page of the workspace editor and make sure the question mark button is shown in the header. 
2. This includes every page of the bank account setup flow (info on how to test the bank setup flow [here](https://stackoverflow.com/c/expensify/questions/342/525#525)) 
3. Click that button on each page and make sure you are taken to the new Get Assistance page
4. Make sure the task ID section of the URL makes sense given the page you clicked the button from. For example, if you click it from the travel page, you should see the URL as `/get-assistance/WorkspaceBookTravel`
5. Make sure that closing the page closes the entire settings modal
6. Make sure that the back button takes you back to the page you came from
7. Make sure that clicking the `chat with concierge` option navigates you to your Concierge chat
8. Make sure that clicking the `request a setup call` option takes you to the request a call page with the same task ID in the URL (i.e. `/request-call/WorkspaceBookTravel`) 
9. Make sure that you can request a call and that you are navigated back to the page you came from after doing so
10. Check that the API call is using the proper policy ID and task ID in the Network tab


### QA Steps
1. Click through every page of the workspace editor and make sure the question mark button is shown in the header. 
2. This includes every page of the bank account setup flow
3. Click that button on each page and make sure you are taken to the new Get Assistance page
4. Make sure the task ID section of the URL (the last part) makes sense given the page you clicked the button from. For example, if you click it from the travel page, you should see the URL as `/get-assistance/WorkspaceBookTravel`
5. Make sure that closing the page closes the entire settings modal
6. Make sure that the back button takes you back to the page you came from
7. Make sure that clicking the `chat with concierge` option navigates you to your Concierge chat
8. Make sure that clicking the `request a setup call` option takes you to the request a call page with the same task ID in the URL (i.e. `/request-call/WorkspaceBookTravel`) 
9. Make sure that you can request a call and that you are navigated back to the page you came from after doing so
10. Make sure you receive an sms saying "Thanks for requesting a call with our Expensify Guides". 

<!--- 
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image 
--->

### Tested On

- [x] Web
- [ ] Mobile Web
- [x] Desktop
- [ ] iOS
- [x] Android

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->
<img width="1370" alt="Screen Shot 2022-01-03 at 2 11 15 PM" src="https://user-images.githubusercontent.com/1371996/147986065-906c09f6-683d-4a23-9b4e-583dc0d8bae8.png">

#### Mobile Web
<!-- Insert screenshots of your changes on the web platform (from a mobile browser)-->

#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->
<img width="1200" alt="Screen Shot 2022-01-03 at 3 28 42 PM" src="https://user-images.githubusercontent.com/1371996/147991306-1d1a8274-c18d-4d0f-bcf6-6a612b68d888.png">

#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->

<img width="360" alt="Screen Shot 2022-01-03 at 3 59 03 PM" src="https://user-images.githubusercontent.com/49007721/147993047-639ffb49-cbe7-45a0-8539-9ff79186a114.png">
<img width="360" alt="Screen Shot 2022-01-03 at 3 59 05 PM" src="https://user-images.githubusercontent.com/49007721/147993051-d4b32bd9-5e09-4a3a-8f66-bf2fc0f97504.png">

#### Android
<!-- Insert screenshots of your changes on the Android platform-->
<img width="335" alt="Screen Shot 2022-01-03 at 2 35 48 PM" src="https://user-images.githubusercontent.com/1371996/147987927-628ce485-074e-4906-94fb-ddc10a4343c8.png">
<img width="324" alt="Screen Shot 2022-01-03 at 2 36 03 PM" src="https://user-images.githubusercontent.com/1371996/147987946-09bb7fb8-75d8-4bd9-ab47-68f37ed3f373.png">

puneetlath committed:
Rename unused shouldShowInboxCallButton prop to shouldShowGetAssistanceButton

puneetlath committed:
Set prop to show get assistance button on all workspace pages

puneetlath committed:
Add copy for get assistance page

puneetlath committed:
Replace InboxCallButton with GetAssistanceButton in Header

puneetlath committed:
Add shell get assistance page

puneetlath committed:
Add routing to get assistance page

puneetlath committed:
Add back button to get assistance page

puneetlath committed:
Add withLocalize to GetAssistancePage

puneetlath committed:
Add menu items for chat and phone to GetAssistancePage

puneetlath committed:
Add navigate to concierge chat funcionality to GetAssistancePage

puneetlath committed:
Fix issue where navigating to concierge only works every other time

puneetlath committed:
Add navigate to RequestCallPage funcionality to GetAssistancePage

puneetlath committed:
Have back button on RequestCallPage go back to previous screen

puneetlath committed:
Remove deprecated InboxCallButton component

puneetlath committed:
Move GetAssistancePage into workspace folder

puneetlath committed:
Update with proper spanish translations

puneetlath committed:
Update get assistance routing to be within workspace section

puneetlath committed:
Update HeaderWithCloseButton to take policyID as optional prop

puneetlath committed:
Update GetAssistance description spanish translation

puneetlath committed:
Add question mark icon for get assistance

puneetlath committed:
Decouple get assistance route from workspace route

puneetlath committed:
Remove policyID prop from HeaderWithCloseButton

puneetlath committed:
Store last accessed workspace in Onyx whenever workspace is loaded

puneetlath committed:
Add guides call task ID to get assistance route params

puneetlath committed:
Add guides call task IDs as constants

puneetlath committed:
Pass the guides call task ID to header as a prop from all workspace pages

puneetlath committed:
Pass the guides call task ID from get assistance page to request call page

puneetlath committed:
Navigate back instead of to concierge dm after requesting call

puneetlath committed:
Refactor GetAssistancePage into stateless functional component

puneetlath committed:
Update VideoChatButtonAndMenu to use guides call constant instead of hardcoding

puneetlath committed:
If there is a lastAccessedWorkspace in Onyx, use that for the guides call

Luke9389 commented:
I tested this code on all platforms and added screens for iOS.

Luke9389 commented on src/libs/actions/Report.js:
```diff
@@ -1489,8 +1489,8 @@ function navigateToConciergeChat() {
         return;
     }
 
-    Navigation.navigate(ROUTES.getReportRoute(conciergeChatReportID));
     Navigation.closeDrawer();
+    Navigation.navigate(ROUTES.getReportRoute(conciergeChatReportID));
```
I'm noticing that now that these are swapped, we see the 'Get assistance' page close as the chat gets opened with concierge. This creates two sliding pages at once, which feels a bit strange. Did you have a specific reason in mind for this change?

https://user-images.githubusercontent.com/49007721/147994350-1142b1e1-6d85-4d17-a424-c7e7f4acb6f1.mov

puneetlath replied:
Ah, yes. The double animation is indeed a bit jarring. 

The reason I flipped the order was because in the original order there was a bug where it would only actually navigate to the concierge chat every other time you tapped it for some reason.

Luke9389 replied:
Oh... interesting.
That seems like something we should investigate. Perhaps it'd be best to make a new issue for that. 

Swapping the order feels like a workaround rather than a fix. I'd vote to keep the broken behavior around so that it sticks out and we address it.

marcaaron replied:
>in the original order there was a bug where it would only actually navigate to the concierge chat every other time you tapped it for some reason

Are we investigating this anywhere else? Should we mention that the order is significant in the code? Looking at these two lines and nothing about them tells me they are arranged in that way to prevent a bug.

puneetlath replied:
Ok, that's good feedback. I reported it as a bug [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1641323776029200) and I'll revert this change so that it can be handled separately.

puneetlath replied:
Seems there's actually already an issue here: https://github.com/Expensify/App/issues/6908

Luke9389 commented on src/pages/RequestCallPage.js:
```diff
@@ -102,19 +99,23 @@ class RequestCallPage extends Component {
+            ? this.props.lastAccessedWorkspace : _.find(this.props.policies, policy => policy && policy.type === CONST.POLICY.TYPE.PERSONAL);
+
+        if (!policyForCall) {
             Growl.error(this.props.translate('requestCallPage.growlMessageNoPersonalPolicy'), 3000);
```
I read the copy for the `growlMessageNoPersonalPolicy`, and I'm wondering what a user should do in this instance. Should we suggest they go to old dot and make a personal policy?

puneetlath replied:
Yeah it is a bit of an odd scenario. That was something that was already there from before. I'm not sure how a user would even get into that scenario because it shouldn't be possible not to have a personal policy and there isn't a way to create one. It's just something every account has. 

It looks like @jasperhuangg added that [here](https://github.com/Expensify/App/pull/3773), though I'm not sure what the scenario would be where there is no personal policy. Perhaps we could update the error message to recommend that they reach out to Concierge to ask for assistance.

marcaaron replied:
+1 Feels like the user can do nothing in this case and our code is broken somehow. Maybe we should log an alert instead? Why is there no policy?

Separately, it seems totally impossible that they would have... no policies at all. So we should allow this request to be made regardless and then default to the personal policy in the back end if none was provided?

puneetlath replied:
Hm, so I looked into the web code that handles the creation of the job and it seems that everything should work fine even if we don't pass a policy ID. So I'm just going to allow the request to be made in this scenario.

Luke9389 commented on src/pages/workspace/GetAssistancePage.js:
```diff
@@ -0,0 +1,72 @@
+                        <Text style={[styles.h1, styles.flex1]}>
+                            {props.translate('getAssistancePage.subtitle')}
+                        </Text>
+                        <RequestCallIcon width={160} height={100} style={styles.flex1} />
```
I'm not sure about the way this is being imported and used here. 
I think the preferred way to do this is to add that svg to [Illustrations.js](https://github.com/Expensify/App/blob/main/src/components/Icon/Illustrations.js).

Luke9389 replied:
Actually, looping back to this, have you seen/considered using the [WorkspaceSection](https://github.com/Expensify/App/blob/main/src/pages/workspace/WorkspaceSection.js) component? 
Here's an [implementation example.](https://github.com/Expensify/App/blob/main/src/pages/workspace/bills/WorkspaceBillsFirstSection.js#L38-L52)

I think this is the preferred way to build these pages. It would end up encapsulating your MenuItem components below as well.

marcaaron replied:
Is this a "workspace" page + correct to place it in `/pages/workspace/` directory? Asking because it feels a bit more generic. If there are things we can reuse from `WorkspaceSection` that is cool, but we should maybe give it a more generic name unless we are considering this modal screen to be a "workspace" page.

Luke9389 replied:
Yea, that's a good point. This isn't a workspace page per se. `WorkspaceSection` doesn't really seem to have workspace-specific funcitonality/styling. Maybe we should rename it to be more descriptive of what it really is (a title with an image on the side and some menu items). Do you think it's a good idea to rename it and use it here?

marcaaron replied:
>Do you think it's a good idea to rename it and use it here?

Depends on whether we are right about this:

>have you seen/considered using the WorkspaceSection component

Honestly, didn't look at that component to see if it is the same as is suggested - just trying to encourage consistent naming practices here. Feels like there's enough information here for @puneetlath to decide what to do.

puneetlath replied:
>I think the preferred way to do this is to add that svg to Illustrations.js.
Makes sense! I just followed the way it was on the `RequestCallPage` since this part of the page is the exact same. I'll update.

>Is this a "workspace" page + correct to place it in /pages/workspace/ directory? Asking because it feels a bit more generic.

Hm, yeah this is a good question. It's not really a page _of_ the workspace, but it can only be triggered from within a workspace today. I could imagine a future where it could be triggered from outside the workspace as well. I'm thinking that given that it's not specifically a page of the workspace that it makes sense to keep it in the main `/pages` folder instead of the `/pages/workspace` folder. 


>Actually, looping back to this, have you seen/considered using the WorkspaceSection component?
That's interesting. You're right that it would work well for both this page and the `RequestCall` page. Perhaps that's a good reason to make the `WorkspaceSection` component more generic. Perhaps something like `PageSectionWithIcon`

Luke9389 replied:
I like the idea of generalizing the `WorkspaceSection` to be more generic. Could we call it `PageSectionWithIconAndMenu` or is that too verbose? That a menu is included in that component seems worth the mention. Unless there's a notion of a menu always being present in a "page section".

puneetlath replied:
Ok, I've gone ahead and moved this component out of the Workspace folder and made it a more generic `Section` component.

Luke9389 reviewed (changes requested):
These changes are looking great so far! Excellent job here. 
I've left some suggestions and questions, some of which are more about UX than your specific code (though there are a few code recommendations as well). 

I'm going to request another pair of 👀 on this as well for good measure.

marcaaron commented on src/libs/Navigation/linkingConfig.js:
```diff
@@ -125,6 +125,9 @@ export default {
                     },
+                    GetAssistance: {
+                        path: ROUTES.GET_ASSISTANCE,
+                    },
```
Is the `Settings` modal stack the right spot for this? Or should the Get Assistance page have its own modal stack dedicated to it?

puneetlath replied:
Yep, as we discussed [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1641854769243700), the settings stack is the right place for this since it can only be accessed from within the workspace and that's where we want the back button to go back to.

marcaaron commented on src/libs/actions/Inbox.js:
```diff
@@ -29,7 +27,7 @@ function requestInboxCall({
         .then((result) => {
             if (result.jsonCode === 200) {
                 Growl.success(Localize.translateLocal('requestCallPage.growlMessageOnSave'));
-                Report.fetchOrCreateChatReport([email, CONST.EMAIL.CONCIERGE], true);
```
Were we only calling `fetchOrCreateChatReport()` here to navigate to the Concierge chat? Or did we also need to sometimes create it?

puneetlath replied:
I don't see why we'd need to create it in this scenario anymore. We'll take you back to whatever page you came from after the call is requested. We aren't navigating you to your Concierge chat here, so I think we're good.

marcaaron commented on src/libs/actions/Policy.js:
```diff
@@ -403,6 +403,13 @@ function hideWorkspaceAlertMessage(policyID) {
+ * @param {String} policyID
+ */
+function updateLastAccessedWorkspace(policyID) {
+    Onyx.set('lastAccessedWorkspace', policyID);
```
Should be using `ONYXKEYS` constant here?

puneetlath replied:
Good call. Updated.

marcaaron commented on src/pages/RequestCallPage.js:
```diff
@@ -102,19 +99,23 @@ class RequestCallPage extends Component {
 
-        const personalPolicy = _.find(this.props.policies, policy => policy && policy.type === CONST.POLICY.TYPE.PERSONAL);
-        if (!personalPolicy) {
+        // If there's a lastAccessedWorkspace in Onyx then use that as the policy for the call, otherwise use the personal policy
```
This comment can probably be removed or improved. It describes what the code is doing but would be better if it could perhaps explain **why** a `policyID` is required to request an inbox call or **why** we'd prefer the workspace over the personal policy.

puneetlath replied:
I removed it. With the updated way I'm setting this, I don't think a comment is necessary.

marcaaron commented on src/pages/RequestCallPage.js:
```diff
@@ -102,19 +99,23 @@ class RequestCallPage extends Component {
-        if (!personalPolicy) {
+        // If there's a lastAccessedWorkspace in Onyx then use that as the policy for the call, otherwise use the personal policy
+        const policyForCall = this.props.lastAccessedWorkspace
+            ? this.props.lastAccessedWorkspace : _.find(this.props.policies, policy => policy && policy.type === CONST.POLICY.TYPE.PERSONAL);
```
Just a thought, but what if we change this to guarantee a `policy` is returned like so? 

```js
const policyForCall =_.find(this.props.policies, policy => {
    if (!policy) {
        return;
    }
    
    if (this.props.lastAccessedWorkspacePolicyID) {
        return policy.id === this.props.lastAccessedWorkspacePolicyID;
    }

    return policy.type === CONST.POLICY.TYPE.PERSONAL;
});
```

Then we won't need line `111` and also the `policyForCall` variable will no longer ambiguously refer to either a "policy" or "policyID" type?

puneetlath replied:
smart, thanks!

marcaaron commented on src/pages/RequestCallPage.js:
```diff
@@ -206,10 +207,7 @@ class RequestCallPage extends Component {
-                        onBackButtonPress={() => Report.fetchOrCreateChatReport([
-                            this.props.session.email,
-                            CONST.EMAIL.CONCIERGE,
-                        ], true)}
```
Left a similar comment elsewhere, but just want to be certain that we were not depending on a chat to be created at this point. Probably we are not and this is fine.

puneetlath replied:
Nope, we weren't depending on it before or now. Before, the call could only be requested from the Concierge DM, so it was guaranteed to already exist. Now we are taking you back to whatever page you came from instead.

marcaaron reviewed (changes requested):
Nice changes here overall! Left a few improvement ideas.

puneetlath committed:
Revert fix to issue where navigating to concierge only works every other time

puneetlath commented on src/ONYXKEYS.js:
```diff
@@ -172,4 +172,6 @@ export default {
     WALLET_TRANSFER: 'walletTransfer',
 
+    // The last workspace whose settings were accessed by the user
+    LAST_ACCESSED_WORKSPACE: 'lastAccessedWorkspace',
```
That's good feedback! I'll udpate.

puneetlath committed:
Update Onyx key from lastAccessedWorkspace to lastAccessedWorkspacePolicyID

puneetlath committed:
Add concierge exclamation illustration

puneetlath committed:
Refactor WorkspaceSection component to be more generic PageSectionWithIcon

puneetlath committed:
Move GetAssistancePage out of workspace folder to main pages folder

puneetlath committed:
Refactor GetAssistancePage to use PageSectionWithIcon

puneetlath committed:
Rename PageSectionWithIcon to Section

puneetlath committed:
Refactor RequestCallPage to use Section

puneetlath committed:
Improve the code for getting the policy ID for Guides call

puneetlath committed:
Remove no longer used growlMessageNoPersonalPolicy

puneetlath committed:
Merge branch 'main' into puneet-workspace-assistance

puneetlath committed:
Merge branch 'main' into puneet-workspace-assistance

puneetlath committed:
Fix personal details error messages I messed up on merge

puneetlath commented:
Ok @Luke9389 @marcaaron I'm ready for another review. Main things that have been updated: 

1. I improved the way we were setting the policy ID for the call to be more straightforward
2. I refactored the `WorkspaceSection` component to be a more generic Section component that can be used outside the workspace
3. I used that `Section` component on the new `GetAssistance` page and updated the `RequestCall` page to use it as well

marcaaron commented on src/components/Section.js:
```diff
@@ -31,7 +31,7 @@ const defaultProps = {
 };
 
-const WorkspaceSection = (props) => {
+const Section = (props) => {
```
👌

marcaaron commented on src/pages/GetAssistancePage.js:
```diff
@@ -0,0 +1,67 @@
+                menuItems={[
+                    {
+                        title: props.translate('getAssistancePage.chatWithConcierge'),
+                        onPress: () => (Report.navigateToConciergeChat()),
```
NAB, these parens are not necessary.

marcaaron commented on src/pages/GetAssistancePage.js:
```diff
@@ -0,0 +1,67 @@
+                    },
+                    {
+                        title: props.translate('getAssistancePage.requestSetupCall'),
+                        onPress: () => (Navigation.navigate(ROUTES.getRequestCallRoute(props.route.params.taskID))),
```
Same here. The parens are not required.

marcaaron commented on src/pages/RequestCallPage.js:
```diff
@@ -69,12 +64,15 @@ const propTypes = {
         loading: PropTypes.bool,
     }),
+
+    lastAccessedWorkspacePolicyID: PropTypes.string,
```
Needs a doc.

puneetlath replied:
Nice catch, added!

marcaaron reviewed (changes requested):
Nice work! Just have nabs and there is one missing doc for propTypes

Luke9389 commented on src/ONYXKEYS.js:
```diff
@@ -172,4 +172,6 @@ export default {
     // Stores information about active wallet transfer amount, selectedAccountID, status, etc
     WALLET_TRANSFER: 'walletTransfer',
 
+    // The policy id of the last workspace whose settings were accessed by the user
```
NAB: Could say `policyID` here to make it searchable.

Luke9389 commented on src/components/HeaderWithCloseButton.js:
```diff
@@ -63,8 +64,8 @@ const propTypes = {
 
-    /** The task ID to associate with the call button, if we show it */
-    inboxCallTaskID: PropTypes.string,
+    /** The guides call task ID to associate with the get assistance button, if we show it */
```
same here w/ `taskID`

Luke9389 commented on src/pages/GetAssistancePage.js:
```diff
@@ -0,0 +1,67 @@
+                    },
+                ]}
+            >
+                <View style={[styles.mv4]}>
```
NAB: This can be `style={styles.mv4}`

Luke9389 commented on src/pages/RequestCallPage.js:
```diff
@@ -206,41 +209,39 @@ class RequestCallPage extends Component {
+                            title={this.props.translate('requestCallPage.subtitle')}
+                            icon={Illustrations.ConciergeExclamation}
+                        >
+                            <Text style={[styles.mb4]}>
```
NAB: Same here, can be `style={styles.mb4}`

Luke9389 reviewed (dismissed):
Nice! All NABs from me too. Great job!

puneetlath committed:
Minor style fixes

puneetlath commented on src/pages/RequestCallPage.js:
```diff
@@ -69,12 +64,16 @@ const propTypes = {
         loading: PropTypes.bool,
     }),
+
+    /** The policy id of the last workspace whose settings the user accessed */
```
@marcaaron added the doc here.

puneetlath commented:
Thanks for the feedback y'all! @marcaaron I added the doc you requested if you can give another review.

puneetlath committed:
Make policy id in comment policyID to make it more searchable

puneetlath commented on src/pages/RequestCallPage.js:
```diff
@@ -69,12 +64,16 @@ const propTypes = {
         loading: PropTypes.bool,
     }),
+
+    /** The policyID of the last workspace whose settings the user accessed */
```
@marcaaron the doc is here.

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1685977851) to staging by @marcaaron in version: 1.1.27-2 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1687245820) to staging by @marcaaron in version: 1.1.27-3 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1687245820) to staging by @marcaaron in version: 1.1.27-3 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|skipped 🚫 
🖥 desktop 🖥|skipped 🚫
🍎 iOS 🍎|skipped 🚫 
🕸 web 🕸|skipped 🚫

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1699278343) to production by @Jag96 in version: 1.1.29-5 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
